cdap-docs/developers-manual/source/data-exploration/streams.rst

```diff
@@ -8,82 +8,82 @@
 Stream Exploration
 ==================
 
-Streams are the primary method of ingesting real-time data into CDAP.
-It is often useful to be able to examine data in a stream in an ad-hoc manner through
-SQL-like queries
+Streams are the primary method of ingesting real-time data into CDAP. It is often useful
+to be able to examine data in a stream in an ad-hoc manner through SQL-like queries
 
 Each event in a stream contains a timestamp, a map of headers, and a body. When a stream
-is created, a corresponding Hive table is created that allows queries to be run over
-those three columns. Many times, stream event bodies are also structured and have
-a format and schema of their own. For example, event bodies may be comma-delimited
-text or Avro-encoded binary data. In those cases, it is possible to set a format and schema
-on a stream, enabling more powerful queries.
+is created, a corresponding Hive table is created that allows queries to be run over those
+three columns. Many times, stream event bodies are also structured and have a format and
+schema of their own. For example, event bodies may be a format such as comma-delimited
+text or Avro-encoded binary data. In those cases, it is possible to set a format and
+schema on a stream, enabling more powerful queries.
+
+Let's take a closer look at attaching formats and schemas on streams.
 
-Let's take a closer look at attaching format and schema on streams.
 
 .. _stream-exploration-stream-format:
 
 Formats
--------
+=======
+
+A **format** defines how the bytes in an event body can be read as a higher-level object. For
+example, the CSV (comma-separated values) format can read each value in comma-delimited
+text as a separate column of some given type.
 
-A format defines how the bytes in an event body can be read as a higher level object.
-For example, the CSV (comma separated values) format can read each value in comma-delimited text
-as a separate column of some given type. Each format has
-a schema that describes the structure of data the format can read. Some formats, such as the Avro format,
-require a schema to be explicitly given. Other formats, such as the CSV format, have a default schema.
+Each format has a schema that describes the structure of data the format can read. Some
```

Comment:
Format doesn't always have a schema. CLF does define format and a schema, but not true for all. CSV is just a format and there is no schema for it.